tests/admin-mode_test.go

```diff
@@ -0,0 +1,270 @@
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "BDD test")
+}
+
+var _ = BeforeSuite(func() {
+	var err error
+	kubeconfig = os.Getenv("HOME") + "/.kube/config"
+	config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get config")
+	}
+
+	client, err = kubernetes.NewForConfig(config)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get client")
+	}
+
+	clientSet, err = chaosClient.NewForConfig(config)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get clientSet")
+	}
+
+	err = v1alpha1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	//Checking the status of operator which is already installed and running
+	operator, err := client.AppsV1().Deployments(chaosTypes.ChaosNamespace).Get("litmus", metav1.GetOptions{})
+	Expect(err).To(BeNil(), "Failed to get operator")
+	count := 0
+	for operator.Status.UnavailableReplicas != 0 {
+		if count < 50 {
+			fmt.Printf("Unavaliable Count: %v \n", operator.Status.UnavailableReplicas)
+			operator, err = client.AppsV1().Deployments(chaosTypes.ChaosNamespace).Get("litmus", metav1.GetOptions{})
+			Expect(err).To(BeNil(), "Failed to get operator")
+			time.Sleep(5 * time.Second)
+			count++
+		} else {
+			Fail("Operator is not in Ready state Time Out")
+		}
+	}
+	fmt.Println("Chaos Operator is up and Running")
+})
+
+//BDD Tests for testing admin mode functionality
+//Checking for the creation of runner pod for application in same namespace
+var _ = Describe("BDD of operator reconcile resiliency check", func() {
+
+	// BDD TEST CASE 1
+	// All Litmus components are in litmus namespace and application pod is in defalt ns
+	Context("Check for litmus components", func() {
+
+		It("Should check for creation of runner pod", func() {
+
+			//Creating application for pod-delete in test namespace
+			By("Creating deployment for pod-delete chaos")
+			err = exec.Command("kubectl", "run", "adminapp", "--image=nginx").Run()
+			Expect(err).To(BeNil(), "Failed to create deployment")
+			fmt.Println("Test Application is created")
+
+			//Removing all Experiment if it allready exists in litmus ns
+			By("Experiment Cleanup")
+			err = exec.Command("kubectl", "delete", "chaosexperiment", "pod-delete", "-n", chaosTypes.ChaosNamespace).Run()
+			Expect(err).To(BeNil(), "Failed to delete chaosexperiment")
+			fmt.Println("Chaos experiment deleted successfully")
+
+			//Removing all chaosengines in litmus namespace
+			By("Experiment Cleanup")
+			err = exec.Command("kubectl", "delete", "chaosengine", "engine", "-n", chaosTypes.ChaosNamespace).Run()
+			Expect(err).To(BeNil(), "Failed to delete chaosengine")
+			fmt.Println("chaos engine deleted successfully")
+
+			//Installing RBAC for experiment that is pod-delete
+			rbacPath := "https://raw.githubusercontent.com/litmuschaos/pages/master/docs/litmus-admin-rbac.yaml"
+			err = exec.Command("wget", "-O", experimentName+"-sa.yaml", rbacPath).Run()
```

Comment:
this rbac is for all experiments so you can use a generic name say admin-rbac.yaml ?

Reply:
pod-delete-sa.yaml file is already there, as you already created that during pod-delete exp

Reply:
np... it will modify that file anyways it is tmp file.
have tested in that way only.